### PR TITLE
adapter: add cluster_id to audit log create events

### PR DESCRIPTION
**Pull request by SangJunBak**
### Motivation

  * This PR adds a feature that has not yet been specified.
  For displaying freshness in the Console grouped by cluster, we need to know an object's cluster ID even when it's dropped. This allows us to know the cluster ID of a dropped object.

Context: https://materializeinc.slack.com/archives/C073FCHFWV6/p1726082674974979

### Checklist

- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

I feel like this isn't a major user-facing behavior change because
a) it's backwards compatible
b) we don't document the possible fields for audit_event.details

### CODE REVIEW

**Review by benesch — COMMENTED** [review]
Deferring explicit approval to @ParkMyCar, but I took a quick look and this largely LGTM!

File: src/catalog/protos/hashes.json [hunk 1]
```diff
@@ -21,6 +21,6 @@
   },
   {
     "name": "objects_v64.proto",
-    "md5": "e914c3020f9625f33d2ac89579020faf"
+    "md5": "35ed48a2d63a9c17d14b3ed2c648abca"
```
Comment (ParkMyCar) [inline src/catalog/protos/hashes.json:24]: _Technically_ your change to the protobuf files is forwards and backwards compatible, but the way we've structured the catalog migration is the hashes for the `object_X.proto` files should never change. The goal is they exist as a snapshot in time of what was serialized to the catalog.

If you don't mind refactoring this a bit, can you create a new `objects_v65.proto` that has your changes, and leave `objects_v64.proto` untouched? Then add `v65` to the [`objects!`](https://github.com/MaterializeInc/materialize/blob/main/src/catalog/src/durable/upgrade.rs#L184-L191) macro, and bump the `CATALOG_VERSION: u64 = 65`. Finally, add an upgrade [here](https://github.com/MaterializeInc/materialize/blob/main/src/catalog/src/durable/upgrade.rs#L288) for v64, and it'll just be a no-op upgrade like [v63_to_v64](https://github.com/MaterializeInc/materialize/blob/main/src/catalog/src/durable/upgrade/v63_to_v64.rs).

I know this is a lot of boilerplate for a change that should be quite easy, but by sticking to this framework it's made it easier to reason about how the catalog evolves. More than happy to do some pair programming here if it helps!

**Review by ParkMyCar — CHANGES_REQUESTED** [review]

File: src/catalog/protos/hashes.json (hunk 1 above)
Comment (SangJunBak) [reply src/catalog/protos/hashes.json:24]: Yeah for sure!

**Comment by SangJunBak** [comment]
@ParkMyCar Thanks for the thorough instructions! Amended the commit with the new changes

File: src/catalog/protos/hashes.json [hunk 2]
```diff
@@ -1,10 +1,11 @@
 [
   {
     "name": "objects.proto",
-    "md5": "b09920c5b66ac584bcefdf361e769e1f"
+    "md5": "497b292990f570bc2f75586d35dbdada"
   },
   {
     "name": "objects_v60.proto",
+
```
Comment (ParkMyCar) [inline src/catalog/protos/hashes.json:8]: nit: remove the newline

**Review by ParkMyCar — APPROVED** [review]
Sweeeeeet, thanks @SangJunBak!

**Pull request merged**